test/constraints/constraint_unit.jl

```diff
@@ -947,6 +1017,92 @@ function test_constraint_min_up_time()
+                repeat([stochastic_scenario(:parent)], tail_hours)
+            ]
+            time_slices = [
+                reverse(time_slice(m; temporal_block=temporal_block(:hourly)))
+                reverse(history_time_slice(m; temporal_block=temporal_block(:hourly)))
+            ][1:(head_hours + tail_hours)]
+            @testset for h in 1:length(constraint)
+                s_set, t_set = scenarios[h:(h + tail_hours - 1)], time_slices[h:(h + tail_hours - 1)]
+                s, t = s_set[1], t_set[1]
+                path = reverse(unique(s_set))
+                var_u_oos_key = (unit(:unit_ab), s, t)
+                var_u_oos = var_units_out_of_service[var_u_oos_key...]
+                vars_u_toos = [var_units_taken_out_of_service[unit(:unit_ab), s, t] for (s, t) in zip(s_set, t_set)]
+                expected_con = @build_constraint(var_u_oos >= sum(vars_u_toos))
+                con_key = (unit(:unit_ab), path, t)
+                observed_con = constraint_object(constraint[con_key...])
+                @test _is_constraint_equal(observed_con, expected_con)
+            end
+        end
+    end
+end
+
+
+function test_constraint_min_scheduled_outage_duration()
+    @testset "constraint_min_scheduled_outage_duration" begin
+        model_end = Dict("type" => "date_time", "data" => "2000-01-01T05:00:00")
+        @testset for scheduled_outage_duration_minutes in (60, 120, 210)
+            url_in = _test_constraint_unit_setup()
+            scheduled_outage_duration = Dict("type" => "duration", "data" => string(scheduled_outage_duration_minutes, "m"))
+            object_parameter_values = [
+                ["unit", "unit_ab", "scheduled_outage_duration", scheduled_outage_duration],
+                ["unit", "unit_ab", "outage_variable_type", "unit_online_variable_type_integer"],
+                ["model", "instance", "model_end", model_end],                
+            ]
+            SpineInterface.import_data(url_in; object_parameter_values=object_parameter_values)
+            m = run_spineopt(url_in; log_level=0, optimize=false)
+            var_units_out_of_service = m.ext[:spineopt].variables[:units_out_of_service]            
+            constraint = m.ext[:spineopt].constraints[:min_scheduled_outage_duration]
+            constraint_t = current_window(m)
+            @test length(constraint) == 1
+            s_path=[stochastic_scenario(:parent), stochastic_scenario(:child)]
+            
+            scenarios = [
+                [stochastic_scenario(:parent)] 
+                repeat([stochastic_scenario(:child)], 4)
+            ]
+            time_slices = time_slice(m; temporal_block=temporal_block(:hourly))
+
+            vars_u_oos = [var_units_out_of_service[unit(:unit_ab), s, t] for (s, t) in zip(scenarios, time_slices)]
+            expected_con = @build_constraint(sum(vars_u_oos) >= scheduled_outage_duration_minutes/60 )
+            @show expected_con
```

Comment:
Maybe remove the @show?

Reply:
Thanks, done in [latest commit](https://github.com/spine-tools/SpineOpt.jl/commit/21097e41186dd36e83125e591bcfd34989910eeb)